mitodl/open-discussions
# topic selector UI component

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #2929
closes #2938

#### What's this PR do?

this adds a fancier way to deal with the 'subject' filter - basically, we have a text input that filters down the list based on what the user types. so they can type in 'bio' and get things like 'Biology', 'Biochemistry', 'Biomedical Engineering', etc.

#### How should this be manually tested?

go to the course search and play around with it. you should be able to filter down the subjects or scroll through a list of all of them. the list of subjects should change in response to other options being selected in the same way as normal. other course search features should all be unchanged.

#### Screenshots (if appropriate)

![Screenshot from 2020-05-29 11-25-05](https://user-images.githubusercontent.com/6207644/83276768-1e648e80-a19f-11ea-9966-41367af66422.png)
![Screenshot from 2020-05-29 11-24-55](https://user-images.githubusercontent.com/6207644/83276770-1e648e80-a19f-11ea-9c3f-44c36554c9d4.png)
![Screenshot from 2020-05-29 11-24-32](https://user-images.githubusercontent.com/6207644/83276771-1efd2500-a19f-11ea-8ad0-93b1972181b7.png)
![Screenshot from 2020-05-29 11-24-20](https://user-images.githubusercontent.com/6207644/83276773-1efd2500-a19f-11ea-9d69-da5bcfbe7b51.png)
